test/test_scheduler.go

```diff
@@ -353,6 +353,97 @@ func (s *SchedulerSuite) TestTCPApp(t *c.C) {
 	}
 }
 
+func (s *SchedulerSuite) TestRollbackController(t *c.C) {
+	// get the current controller release
+	client := s.controllerClient(t)
+	app, err := client.GetApp("controller")
+	t.Assert(err, c.IsNil)
+	release, err := client.GetAppRelease(app.ID)
+	t.Assert(err, c.IsNil)
+
+	watcher, err := s.controllerClient(t).WatchJobEvents(app.ID, release.ID)
+	t.Assert(err, c.IsNil)
+	defer watcher.Close()
+
+	// get the current controller formation
+	formation, err := client.GetFormation(app.ID, release.ID)
+	t.Assert(err, c.IsNil)
+
+	currentReleaseID := release.ID
+
+	// create a controller deployment what will fail
+	release.ID = ""
+	worker := release.Processes["web"]
+	worker.Entrypoint = []string{"/i/dont/exist"}
+	release.Processes["worker"] = worker
+	t.Assert(client.CreateRelease(release), c.IsNil)
+	deployment, err := client.CreateDeployment(app.ID, release.ID)
+	t.Assert(err, c.IsNil)
+
+	events := make(chan *ct.DeploymentEvent)
+	eventStream, err := client.StreamDeployment(deployment, events)
+	t.Assert(err, c.IsNil)
+	defer eventStream.Close()
+
+	// wait for the deploy to complete (this doesn't wait for specific events
+	// due to the fact that when the deployer deploys itself, some events will
+	// not get sent)
+loop:
+	for {
+		select {
+		case e, ok := <-events:
+			if !ok {
+				t.Fatal("unexpected close of deployment event stream")
+			}
+			debugf(t, "got deployment event: %s %s", e.JobType, e.JobState)
+			switch e.Status {
+			case "complete":
+				t.Fatal("the deployment succeeded when it should have failed")
+			case "failed":
+				break loop
+			}
+		case <-time.After(2 * time.Minute):
+			t.Fatal("timed out waiting for the deploy to complete")
+		}
+	}
+
+	// wait for jobs to come back up
+	hosts, err := s.clusterClient(t).Hosts()
+	expected := map[string]map[string]int{
+		currentReleaseID: {
+			"web":       formation.Processes["web"],
+			"worker":    formation.Processes["worker"],
+			"scheduler": len(hosts),
+		},
```

Comment:
Ah ok, for some reason I read `worker` as `web` earlier :unamused: 

Ok so in that case do we expect any new worker jobs to start? Shouldn't the old ones still be intact?